packages/tools/examples/videoTools/index.ts

```diff
@@ -92,74 +117,357 @@ addButtonToToolbar({
+addButtonToToolbar({
+  id: 'AddSelection',
+  title: 'Add',
+  onClick() {
+    viewport.pause();
+    if (!selectedAnnotation.annotationUID) {
+      return;
+    }
+    const newSelection = setSelection(
+      selectedAnnotation.annotationUID,
+      null,
+      viewport.getFrame()
+    );
+    updateAnnotationDiv(selectedAnnotation.annotationUID, newSelection);
+    fireUpdateEvent();
   },
 });
 
-const scrollSpeeds = ['1 f', '2 f', '4 f', '0.5 s', '1 s', '2 s', '4 s'];
+addButtonToToolbar({
+  id: 'SetSelection',
+  title: 'Set',
+  onClick() {
+    viewport.pause();
+    const frame = viewport.getFrame();
+    const { annotationUID, selection } = selectedAnnotation;
+    if (!annotationUID) {
+      return;
+    }
+    const newSelection = setSelection(annotationUID, selection, frame);
+    updateAnnotationDiv(annotationUID, newSelection);
+    fireUpdateEvent();
+  },
+});
 
-const scrollTitle = document.createElement('div');
-scrollTitle.style.display = 'inline';
-scrollTitle.innerText = 'Scroll Distance:';
-toolbar.appendChild(scrollTitle);
+addButtonToToolbar({
+  id: 'Start Range',
+  title: 'Start',
+  onClick() {
+    viewport.pause();
+    const { annotationUID, selection } = selectedAnnotation;
+    const range = toRange(selection);
+    range[0] = viewport.getFrame();
+    if (range[1] < range[0]) {
+      range[1] = range[0];
+    }
+    const updated = setSelection(annotationUID, selection, range);
+    updateAnnotationDiv(annotationUID, updated);
+    fireUpdateEvent();
+  },
+});
 
-addDropdownToToolbar({
-  options: { values: scrollSpeeds, defaultValue: '1 f' },
-  onSelectedValueChange: (value) => {
-    value = value.toString();
-    const unit = value[value.length - 1];
-    const newScrollSpeed = Number(value.substring(0, value.length - 2));
-    viewport.setScrollSpeed(newScrollSpeed, unit);
+addButtonToToolbar({
+  id: 'End Range',
+  title: 'End',
+  onClick() {
+    viewport.pause();
+    const { annotationUID, selection } = selectedAnnotation;
+    const range = toRange(selection);
+    range[1] = viewport.getFrame();
+    if (range[1] < range[0]) {
+      range[0] = range[1];
+    }
+    const updated = setSelection(annotationUID, selection, range);
+    updateAnnotationDiv(annotationUID, updated);
+    fireUpdateEvent();
   },
 });
 
+function annotationSelectionListener(evt) {
+  const { selection } = evt.detail;
+  if (!selection?.length) {
+    selectionDiv.innerHTML = '';
+    return;
+  }
+  const [uid] = selection;
+  updateAnnotationDiv(uid);
+}
+
+function annotationModifiedListener(evt) {
+  updateAnnotationDiv(evt.detail.annotation.annotationUID);
+}
+
+const selectedAnnotation = {
+  annotationUID: '',
+  selection: '',
+};
+
+function toTime(selection) {
+  const range = toRange(selection).map(
+    (it) => Math.round(((it - 1) * 10) / viewport.fps) / 10
+  );
+  if (range[0] === range[1]) {
+    return String(range[0]);
+  }
+  return toSelection(range);
+}
```

Comment:
ideally rename for clarity

function convertSelectionToTimeRange(selection) { ... }
function convertSelectionToNumericRange(selection) { ... }
function navigateAnnotationsRelative(direction) { ... }